prov/efa/src/efa_msg.c

```diff
@@ -97,9 +97,9 @@ static inline ssize_t efa_post_recv(struct efa_base_ep *base_ep, const struct fi
 	}
 
 	wr = &base_ep->efa_recv_wr_vec[wr_index].wr;
-	wr->wr_id = (uintptr_t)msg->context;
 	wr->num_sge = msg->iov_count;
 	wr->sg_list = base_ep->efa_recv_wr_vec[wr_index].sge;
+	wr->wr_id = (uintptr_t) ((flags & FI_COMPLETION) ? msg->context : NULL);
```

Comment:
I discussed with Jessie. According to the docs for FI_CONTEXT

>This structure should be treated as opaque to the application. For performance reasons, this structure must be allocated by the user, but may be used by the fabric provider to track the operation.

So we can use this buffer however we want. I would suggest setting it up as a bitmask and have FI_COMPLETION be one of the bits. That way, we can use the other bits for something else in the future.